huaweicloud/services/sdrs/resource_huaweicloud_sdrs_replication_pair.go

```diff
@@ -0,0 +1,198 @@
+package sdrs
+
+import (
+	"context"
+	"strings"
+	"time"
+
+	"github.com/hashicorp/go-multierror"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/diag"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+
+	"github.com/chnsz/golangsdk/openstack/sdrs/v1/replications"
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/common"
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/config"
+)
+
+func ResourceSDRSReplicationPair() *schema.Resource {
+	return &schema.Resource{
+		CreateContext: resourceSDRSReplicationPairCreate,
+		ReadContext:   resourceSDRSReplicationPairRead,
+		UpdateContext: resourceSDRSReplicationPairUpdate,
+		DeleteContext: resourceSDRSReplicationPairDelete,
+		Importer: &schema.ResourceImporter{
+			StateContext: schema.ImportStatePassthroughContext,
+		},
+
+		Timeouts: &schema.ResourceTimeout{
+			Create: schema.DefaultTimeout(10 * time.Minute),
+			Delete: schema.DefaultTimeout(10 * time.Minute),
+		},
+
+		Schema: map[string]*schema.Schema{
+			"region": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"group_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"volume_id": {
+				Type:     schema.TypeString,
+				Required: true,
+				ForceNew: true,
+			},
+			"delete_target_volume": {
+				Type:     schema.TypeBool,
+				Optional: true,
+				ForceNew: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Optional: true,
+				ForceNew: true,
+				Computed: true,
+			},
+			"replication_model": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"fault_level": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"status": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"target_volume_id": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+		},
+	}
+}
+
+func resourceSDRSReplicationPairCreate(ctx context.Context, d *schema.ResourceData, meta interface{}) diag.Diagnostics {
+	cfg := meta.(*config.Config)
+	region := cfg.GetRegion(d)
+	client, err := cfg.SdrsV1Client(region)
+	if err != nil {
+		return diag.Errorf("error creating SDRS client: %s", err)
+	}
+	// use for WaitForJobSuccess
+	client.Endpoint = client.ResourceBase
+
+	createOpts := replications.CreateOpts{
+		Name:        d.Get("name").(string),
+		Description: d.Get("description").(string),
+		GroupID:     d.Get("group_id").(string),
+		VolumeID:    d.Get("volume_id").(string),
+	}
+
+	n, err := replications.Create(client, createOpts).ExtractJobResponse()
+	if err != nil {
+		return diag.Errorf("error creating SDRS replication pair: %s", err)
+	}
+
+	createTimeoutSec := int(d.Timeout(schema.TimeoutCreate) / time.Second)
```

Comment:
How about this:

```suggestion
	createTimeoutSec := int(d.Timeout(schema.TimeoutCreate).Seconds())
```
It seems easier to understand.